### PR TITLE
[quantization] test {__init__,from_float} on nnq{,d}.Linear

### PR DESCRIPTION
**Pull request by jamesr66a**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#24364 [quantization] test {__init__,from_float} on nnq{,d}.Linear**

Differential Revision: [D16812543](https://our.internmc.facebook.com/intern/diff/D16812543)

### CODE REVIEW

File: test/test_quantized_nn_mods.py [hunk 1]
```diff
@@ -58,6 +58,10 @@ def test_linear_api(self, batch_size, in_features, out_features, use_bias):
         X = torch.rand(batch_size, in_features).float()
         B = torch.rand(out_features).float() if use_bias else None
         qlinear = nnqd.Linear(in_features, out_features)
+        # Run module with default-initialized parameters.
+        # This tests that the constructor is correct.
+        qlinear(X)
```
Comment (zafartahirov) [inline test/test_quantized_nn_mods.py]: Once this is called, we should check if the initialized weights are of proper dimension. The default constructor path might be different, so the check would be appropriate.

File: test/test_quantized_nn_mods.py [hunk 2]
```diff
@@ -114,6 +118,16 @@ def test_linear_api(self, batch_size, in_features, out_features, use_bias):
         # Test JIT
         self.checkScriptable(qlinear, zip([X], [Z_ref]), check_save_load=True)
 
+        # Test from_float
+        float_linear = torch.nn.Linear(in_features, out_features).float()
```
Comment (zafartahirov) [inline test/test_quantized_nn_mods.py]: I don't think `.float()` here is necessary -- it's the default

**Review by z-a-f — APPROVED** [review]

File: test/test_quantized_nn_mods.py (hunk 2 above)
Comment (jamesr66a) [reply test/test_quantized_nn_mods.py]: NOPE. This is an absolutely dangerous assumption. See the following:

```
import torch

print(torch.Tensor(3).dtype)

torch.set_default_tensor_type('torch.DoubleTensor')

print(torch.Tensor(3).dtype)

```

which outputs

```
torch.float32
torch.float64
```

And indeed, under *all tests* the default scalar type is in fact Double:

https://github.com/pytorch/pytorch/blob/0619b57c4c64be625050b32a6be6ee064b5ba2c3/test/common_utils.py#L45

File: test/test_quantized_nn_mods.py (hunk 1 above)
Comment (zafartahirov) [reply test/test_quantized_nn_mods.py]: resolving -- misunderstood the comment

**Pull request closed without merging**